lib/Directives/Directive.php

```diff
@@ -98,9 +98,41 @@ public function finalize(DocumentNode $document): void
 
     /**
      * Should the following block be passed as a CodeNode?
+     *
+     * You should probably return false from this. If you do,
+     * in most cases (unless you directive allows for some fancy
+     * syntax), you will receive a BlockNode object in processNode().
+     *
+     * See CodeNode for more details.
      */
     public function wantCode(): bool
     {
         return false;
     }
+
+    /**
+     * Can this directive apply to content that is not indented under it?
+     *
+     * Most directives that allow content require that content to be
+     * indented under it. For example:
+     *
+     *      .. note::
+     *
+     *          This is my note! It must be indented.
+     *
+     * But some are allowed to apply to content that is *not* indented:
+     *
+     *      .. class:: align-center
+     *
+     *      I will be a "p" tag with an align-center class
+     *
+     * If your directive allows the "class" directive functionality,
+     * return true from this function. The result is that your
+     * directive's process() method will be called for the next
+     * node after your directive (e.g. a ParagraphNode, ListNode, etc)
+     */
+    public function canApplyToNonBlockContent(): bool
```

Comment:
This reads a bit like may or may not apply, is that really the case?

Otherwise let's maybe rename this to 
```suggestion
    public function appliesToNonBlockContent(): bool
```

Also, I had some doubts about whether "non-block content" was the right name (I now think it is).

To understand that, I dug a bit through the code until I stumbled upon this: https://github.com/doctrine/rst-parser/blob/7907706178f02198a423a907d5aa83dc2a356b70/lib/Parser/LineChecker.php#L57-L64

Now I'm confused… on line 63, `$line` is known to be `''`, so this should just `return true`? It looks like the entire method could be summed up to `return $line === '' || trim($line[0]) === '';`

Reply:
Hey @greg0ire!

Good comments here :).

A) About the naming... I'm fine with your suggestion - I've updated it! :).

B) About `isBlockLine()` - you're totally right. this is a poorly written function. I decided not to touch it previously, but now that we're talking about it, I've updated it to your suggestion. This method has good unit tests... but I added one more test for the `''` use-case (which passed locally with the current code AND after my refactoring to your suggestion).